File: Magic/src/main/java/org/Aio/Herblore/Cleaning.java
package org.Aio.Herblore;

import org.Aio.Task;
import org.powbot.api.Condition;
import org.powbot.api.Random;
import org.powbot.api.Tile;
import org.powbot.api.rt4.*;
import org.powbot.api.rt4.walking.model.Skill;
import org.powbot.api.waiter.SkillExpGainedWaiter;

public class Cleaning extends Task {

    public Cleaning() {
        super();
        this.name = "Cleaning";
    }

    @Override
    public boolean shouldExecute() {
        return Inventory.stream().name("Grimy guam leaf").isNotEmpty() || Inventory.stream().name("Grimy harralander").isNotEmpty()
                || Inventory.stream().name("Grimy irit leaf").isNotEmpty();
    }

    @Override
    public void execute() {
        Game.tab(Game.Tab.INVENTORY);
        for (int i = 0; i < 28; i++) {
            Item herb = Inventory.itemAt(i);
            if (herb.name().contains("Grimy")) {
                //if (Random.nextBoolean()){
                //  System.out.println("Cleaning herbs Int");
                //herb.interact("Clean");
                //Condition.sleep(Random.nextInt(175, 225));
                //else {
                herb.click();
                System.out.println("Cleaning herbs");
                Condition.sleep(Random.nextInt(175, 225));
            }
        }
    }
}

File: untitled2/src/main/kotlin/crafting/Tasks/Cleaning.kt
package crafting.Tasks

import org.Crafting.Task
import org.powbot.api.Condition.wait
import org.powbot.api.rt4.Inventory.stream
import org.powbot.api.rt4.walking.model.Skill
import java.util.concurrent.Callable


class Cleaning : Task() {
    init {
        this.name = "Cleaning"
    }

    fun shouldExecute(): Boolean {
        return stream().name("Grimy guam leaf").isNotEmpty()
    }

    fun execute() {
        val xp = Skill.Herblore.experience()
        val itemName = "Grimy guam leaf"
        val Grimy_guam_leaf = stream().name(itemName).first()
        run {
            //Condition.sleep(205);
            Grimy_guam_leaf.click()
            //Condition.sleep(200);
            Grimy_guam_leaf.nextPoint()
            wait(Callable<Boolean> { xp != Skill.Herblore.experience() }, 150, 5)
        }
    }
}